rosidl_typesupport_opensplice_cpp/include/rosidl_typesupport_opensplice_cpp/Requester.hpp

```diff
@@ -0,0 +1,126 @@
+#ifndef __rosidl_typesupport_opensplice_cpp__Requester__h__
+#define __rosidl_typesupport_opensplice_cpp__Requester__h__
+
+#include <atomic>
+#include <random>
+#include <limits>
+#include <ccpp_dds_dcps.h>
+#include <u_entity.h>
+
+#include <rosidl_typesupport_opensplice_cpp/message_type_support.h>
+#include <rosidl_typesupport_opensplice_cpp/service_type_support.h>
+
+namespace rosidl_typesupport_opensplice_cpp
+{
+
+template<typename RequestT, typename ResponseT>
+class Requester
+{
+  public:
+    Requester(DDS::DomainParticipant_ptr participant, const char * service_name,
+              const char * service_type_name) :
```

Comment:
I wouldn't start changing it right now. A lot of libraries use `const char *` when possible to avoid the "overhead" of std:string. I think as long as we don't need std:string like functionality we should stick to plain `const char *`.